modules/objdetect/samples/barcode.cpp

```diff
@@ -0,0 +1,223 @@
+#include <iostream>
+#include "opencv2/objdetect/barcode.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/highgui.hpp"
+
+using namespace cv;
+using namespace std;
+
+static const Scalar greenColor(0, 255, 0);
+static const Scalar redColor(0, 0, 255);
+static const Scalar yellowColor(0, 255, 255);
+static Scalar randColor()
+{
+    RNG rng = theRNG();
+    return Scalar(rng.uniform(0, 255), rng.uniform(0, 255), rng.uniform(0, 255));
```

Comment:
Without reference to RNG it should return the same values.

Reply:
This line produces undefined behavior due function arguments init order is not specified. Here is an example which prints different values depending on compiler: https://gist.github.com/dkurt/d3021a800ab4e1165fa08ac0cd154d5f

Reply:
Done!